lib/commandCenter/commandCenter.js

```diff
@@ -530,71 +555,62 @@ CommandCenter.prototype.executeCommand = function (cmd, session, metaData, cb) {
 		});
 	}
 
-	var that = this;
-
-	state.archivist.createVault('client', 'client', { state: state }, function () {
-		// execute the command
-
-		logger.debug('Executing user command:', cmdInfo.execPath);
-
-		var mod = cmdInfo.mod;
-
-		var paramList;
+	// Create the client vault, so that events and data may be sent
+	// to the client
+	state.archivist.createVault('client', 'client', { state: state }, () => {
+		// Extract parameters, and return early if the parsing failed
+		let paramList;
 
 		try {
-			paramList = that.buildParamList(cmdInfo.params, cmd.params);
+			paramList = this.buildParamList(cmdInfo.params, cmd.params);
 		} catch (err) {
 			return cb(err);
 		}
 
+		// Debug logging
+		logger.debug
+			.log('Executing user command:', cmdInfo.execPath);
+
+		// Prefix the params list with the state object
```

Comment:
While you're at it, could you pass `state` to `buildParamList` and inject it as the first argument there? `unshift` is very slow, and I don't see why we've never done it in that function instead of unshifting it here.

Reply:
Not sure how that would help. `buildParamList` would need to be refactored perhaps?